src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public abstract class Figure implements FigureArea,DrawFIgure {
+    private String color = new ColorSupplier().getRandomColor();
```

Comment:
The instantiation of `ColorSupplier` and the call to `getRandomColor()` should not be done in the field declaration of the `Figure` class. Consider moving this logic to a constructor or a separate method to make the code more testable and to avoid unnecessary object creation every time a `Figure` object is instantiated.